### PR TITLE
Extract scale appliance

### PR DESCRIPTION
**Pull request by mkasztelnik**
This is a third step after #160 and #162 to split optimizer code and remove it from appliance callbacks.

### CODE REVIEW

**Comment by mkasztelnik** [comment]
@dice-cyfronet/atmo-dev-team please take a look

**Comment by Nuanda** [comment]
@mkasztelnik Marek, just to make sure we understand each other :), I will not, on purpose, review the PRs that perform the "large refactoring" you plan. Since I am not involved in ISMOP, that would be counter productive for the process, to have me complaining about the changes you need in your projects :). I will still try to review other changes to Atmo, of course.

**Comment by mkasztelnik** [comment]
@Nuanda mentioned changes can simplify also the way how you are adding and testing changes in plgrid. I'm planning to remove optimizer invocation from the `Appliance` callbacks - so it also can be interested from your point of view :smile:

File: app/services/atmosphere/cloud/scale_appliance.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
+module Atmosphere
```
Comment (Nuanda) [inline app/services/atmosphere/cloud/scale_appliance.rb]: Please provide a short code documentation. One things that would be useful is to describe shortly how does the `quantity` parameter influence the algorithm.

File: app/services/atmosphere/cloud/scale_appliance.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            if appliance.virtual_machines.count > quantity.abs
+              vms = optimization_strategy.vms_to_stop(-quantity)
+              appl_manager.stop_vms!(vms)
+            else
+              appl_manager.unsatisfied(I18n.t('scale_appliance.to_small_vms'))
+            end
+          end
+          action.log(I18n.t('scale_appliance.end'))
+        else
+          action.log('Scaling not allowed')
```
Comment (Nuanda) [inline app/services/atmosphere/cloud/scale_appliance.rb]: Please be more precise - tell the user that `manual` scaling is not allowed (and try to tell why).

**Comment by Nuanda** [comment]
V. small remarks. :+1:

File: app/services/atmosphere/cloud/scale_appliance.rb (hunk 1 above)
Comment (mkasztelnik) [inline app/services/atmosphere/cloud/scale_appliance.rb]: Done in https://github.com/dice-cyfronet/atmosphere/commit/8748bc671e41fb67a1674c8e01ac53f31bc68de4

File: app/services/atmosphere/cloud/scale_appliance.rb (hunk 2 above)
Comment (mkasztelnik) [inline app/services/atmosphere/cloud/scale_appliance.rb]: Done in https://github.com/dice-cyfronet/atmosphere/commit/1dd8b3d117e2e0a91a582c69b0fc3fff1b98ac22

**Comment by mkasztelnik** [comment]
@dice-cyfronet/atmo-dev-team Since it is only moving code around files if no rejection from your side I will merge it today at 12 AM (@Nuanda your remarks are already implemented)

**Pull request merged**